### PR TITLE
fix: StyledRowSingleSelection: added cursor as pointer

### PR DESCRIPTION
**Pull request by billpereira**
**Summary**

Added on the StyledRowSingleSelection

**Change List (commits, features, bugs, etc)**

bug

**Related Issues**

259

- Closes IBM/carbon-addons-iot-react#NUMBER

**Acceptance Test (how to verify the PR)**

- tests here

### CODE REVIEW

**Comment by scottdickerson** [comment]
Thank you for submitting this.   Can you add the cursor: pointer to all rows (not just the selected row?) . You should be able to conditionally branch on the prop in the styled component..

**Comment by billpereira** [comment]
Sure, every clickable row should have the cursor as pointer?

**Comment by filipecorrea** [comment]
Scott's right. I think you made a mistake here and added it just to the selected row.

**Comment by billpereira** [comment]
i have added to the styled component if it hasRowSelection=single to change cursor to pointer

File: src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx [hunk 1]
```diff
@@ -450,6 +456,7 @@ const TableBodyRow = ({
   ) : (
     <StyledTableRow
       key={id}
+      data-single-select={hasRowSelection}
```
Comment (filipecorrea) [inline src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx]: Wouldn't be simpler if we just propagate `hasRowSelection`?

```suggestion
      hasRowSelection={hasRowSelection}
```

File: src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx [hunk 2]
```diff
@@ -124,6 +124,10 @@ const StyledTableRow = styled(TableRow)`
         }
       }
     }
+    ${props => (props['data-single-select'] === 'single' ? `cursor:pointer` : null)}
```
Comment (filipecorrea) [inline src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx]: If the other comment is correct, then:

```suggestion
    ${props => (props.hasRowSelection === 'single' ? `cursor:pointer` : null)}
```

**Pull request merged**